### PR TITLE
Us053 remove loaded sample

### PR DESCRIPTION
**Pull request by tej99**
- Add ability to search business against trading as as well as the name
- Filter out 'inactive' attributes from search and returned business data

Inactive attributes are attributes that have no collection exercise associated to them. If a Business has no active attributes then the get api should return a 404 even if the business itself is in the database.

https://trello.com/c/IQ8agznp/86-us053-int-remove-a-loaded-sample-13

### CODE REVIEW

**Comment by tej99** [comment]
[US053: INT- Remove a loaded sample (13)](https://trello.com/c/IQ8agznp/86-us053-int-remove-a-loaded-sample-13)

File: ras_party/models/models.py [hunk 1]
```diff
@@ -139,7 +156,11 @@ def _get_attributes_for_collection_exercise(self, collection_exercise_id=None):
                 if attributes.collection_exercise == collection_exercise_id:
                     return attributes
 
-        return self.attributes[-1]
+        try:
+            return next((attributes for attributes in self.attributes if attributes.collection_exercise))
+        except StopIteration:
+            logger.error("No active attributes for business", party_id=self.party_uuid)
+            raise RasError("Business with reference does not exist.", reference=self.business_ref, status=404)
```
Comment (andrewmil) [inline ras_party/models/models.py]: Is this error message correct? In this case the business exists in the system but is no longer linked to an exercise?

**Review by andrewmil — APPROVED** [review]
LGTM :+1:

File: ras_party/models/models.py [hunk 2]
```diff
@@ -139,7 +156,12 @@ def _get_attributes_for_collection_exercise(self, collection_exercise_id=None):
                 if attributes.collection_exercise == collection_exercise_id:
                     return attributes
 
-        return self.attributes[-1]
+        try:
+            return next((attributes for attributes in self.attributes if attributes.collection_exercise))
+        except StopIteration:
+            logger.error("No active attributes for business", party_id=self.party_uuid)
```
Comment (rhiGriff) [inline ras_party/models/models.py:162]: Can we add the status code to this log line please as this will help in the splunk logs.

**Review by emilioward — APPROVED** [review]

**Pull request merged**